notebooks/tests/__init__.py

```diff
@@ -17,17 +17,24 @@
 #
 """Unit tests for demo notebooks."""
 import os
+import shutil
+import site
 import stat
-import subprocess
+import subprocess  # noqa: S404
 import sysconfig
 import tarfile
-import site
+from typing import Any
 from typing import Optional
 
 import requests
 
+try:
+    import pyspark
+
+    _SPARK_VERSION = pyspark.__version__
+except ImportError:
+    _SPARK_VERSION = "3.1.2"
```

Comment:
When/why do we run into this ImportError during tests?

Reply:
its imported from the flink environment which doesn't have spark. I've made it clearer that it shold error out in that situation now